features/step_definitions/catalog_source.rb

```diff
@@ -0,0 +1,21 @@
+Given(/^admin creates "([^"]*)" catalog source with image "([^"]*)"$/) do |cs_name, cs_image|
+  ensure_admin_tagged
+  step %Q/I switch to cluster admin pseudo user/
+  step %Q/I use the "openshift-marketplace" project/
+
+  teardown_add {
+    step %Q/admin ensures "#{cs_name}" catalog_source is deleted from the "openshift-marketplace" project/
+  }
```

Comment:
Perhaps `step %Q/admin ensures "#{cs_name}" catalog_source is deleted from the "openshift-marketplace" project after scenario/`.
Current one also works.

Reply:
yeah, maybe both is ok

Reply:
updated based on your comments